java/maven/src/org/netbeans/modules/maven/newproject/BasicPanelVisual.java

```diff
@@ -562,7 +562,9 @@ void read(WizardDescriptor settings) {
         File projectLocation = (File) settings.getProperty(CommonProjectActions.PROJECT_PARENT_FOLDER); //NOI18N
         if (projectLocation == null || projectLocation.getParentFile() == null || !projectLocation.getParentFile().isDirectory()) {
             projectLocation = ProjectChooser.getProjectsFolder();
-        } 
+        } else {
+            projectLocation = projectLocation.getParentFile();
```

Comment:
This looks like a contradiction to the name of the property in line 562. Could it be, that the property is set to the wrong value?

Reply:
Answering myself, yes this is broken (in `BasicPanelVisual#store`):

```java
        String name = projectNameTextField.getText().trim();
        String folder = createdFolderTextField.getText().trim();
        final File parentFolder = new File(folder);
        
        d.putProperty(CommonProjectActions.PROJECT_PARENT_FOLDER, parentFolder);
```

I would rename `parentFolder` to `targetFolder` and change the last line to `d.putProperty(CommonProjectActions.PROJECT_PARENT_FOLDER, targetFolder.getParentFile());`. But that is untested.

Reply:
The naming caused me considerable confusion. 

Note that your suggestion `d.putProperty("projdir", folder.getParentFile())` produces different and incorrect results. The String value of the property name ...PROJECT_PARENT_FOLDER is "projdir"; and in older areas of code is a literal string rather than a define. Your suggestion, save folder.getParentFile() as property value was actually my first attempt to fix this. Things got weird and changes started radiating; reverted that attempt. The property "projdir" is globally used by the wizard infrastructure and it means the base directory of the new project and not the parent directory of the newly created project directory.

After a circuitous route, I decided to find the the screen for a plain old java project. It is 
`java/java.j2seproject/src/org/netbeans/modules/java/j2seproject/ui/wizards/PanelProjectLocationVisual.java`
and it uses the same names and has similar code to its maven counterpart. (Undoubtedly the prototype for this screen)

So, while it's tempting to rename things so they make sense. I am hesitant to diverge from the original code. BTW, in the code you quote from the store() method, renaming the variable `parentFolder` to `projectDirectory` would be accurate. Then when the two lines are restored it would be like saying `projectLocation = projectDirectory.getParentFile()`. It could have read like that, but the original code was probably avoiding declaring new variables. Note that in the code in read() `projectLocation = ProjectChooser.getProjectsFolder()` retrieves the default value for where to put new projects the other branch of the "if" statement is working with the value of "projdir" and does ["projdir"].getParentFile().

Reply:
Ok - I might be wrong, but this is awfully lot text to say, that the project parent folder is not the project parent folder. We should at least try to get this straight, there might be more people like me, that read variables and think that they hold what they say. Sorry I can't say this without less irony.

Reply:
I'm itching to clean this up, glad you brought it up again. Yes, verbose; just saying `123 != 123` seemed inadequate. And I want to record the situation so I don't have to reconstruct it again. It does seem that simply restoring two lines that were mistakenly removed is consistent with a minimal diff philosophy.

PROJECT_PARENT_FOLDER should be renamed, maybe PROJECT_BASE_FOLDER (or just PROJECT_FOLDER); but that's an API change. PARENT could be deprecated in favor of BASE.

How abut this one? (keeping it local)